examples/40_advanced/example_text_preprocessing.py

```diff
@@ -1,79 +1,98 @@
+    1. https://scikit-learn.org/stable/tutorial/text_analytics/working_with_text_data.html
+    2. https://machinelearningmastery.com/clean-text-machine-learning-python/
 
-For processing text features you need a pandas dataframe and set the desired
-text columns to string and the categorical columns to category.
 
-*auto-sklearn* text embedding creates a bag of words count.
 """
+from pprint import pprint
+
+import pandas as pd
+from sklearn.datasets import fetch_20newsgroups
 import sklearn.metrics
-import sklearn.datasets
+
 import autosklearn.classification
 
 ############################################################################
 # Data Loading
 # ============
 
-X, y = sklearn.datasets.fetch_openml(data_id=40945, return_X_y=True)
-
-# by default, the columns which should be strings are not formatted as such
-print(f"{X.info()}\n")
+newsgroups_train = fetch_20newsgroups(subset="train")
+newsgroups_test = fetch_20newsgroups(subset="test")
+
+# load train data
+df_train = pd.DataFrame({"X": [], "y": []})
+
+for idx, (text, target) in enumerate(
+    zip(newsgroups_train.data, newsgroups_train.target)
+):
+    df_train = pd.concat(
+        [
+            df_train,
+            pd.DataFrame(
+                {"X": text, "y": newsgroups_train.target_names[target]}, index=[idx]
+            ),
+        ]
+    )
+
+# explicitly label text column as string
+X_train = df_train.astype({"X": "string", "y": "category"})
+y_train = X_train.pop("y")
+
+# load test data
+df_test = pd.DataFrame({"X": [], "y": []})
+
+for idx, (text, target) in enumerate(zip(newsgroups_test.data, newsgroups_test.target)):
+    df_test = pd.concat(
+        [
+            df_train,
+            pd.DataFrame(
+                {"X": text, "y": newsgroups_train.target_names[int(target)]},
+                index=[idx],
+            ),
+        ]
+    )
+
+# explicitly label text column as string
+X_test = df_test.astype({"X": "string", "y": "category"})
+y_test = X_test.pop("y")
 
-# manually convert these to string columns
-X = X.astype(
-    {
-        "name": "string",
-        "ticket": "string",
-        "cabin": "string",
-        "boat": "string",
-        "home.dest": "string",
-    }
-)
-
-# now *auto-sklearn* handles the string columns with its text feature preprocessing pipeline
-
-X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(
-    X, y, random_state=1
-)
 
-cls = autosklearn.classification.AutoSklearnClassifier(
-    time_left_for_this_task=30,
-    # Bellow two flags are provided to speed up calculations
-    # Not recommended for a real implementation
-    initial_configurations_via_metalearning=0,
-    smac_scenario_args={"runcount_limit": 1},
+############################################################################
+# Build and fit a classifier
+# ==========================
+
+automl = autosklearn.classification.AutoSklearnClassifier(
+    # set the time high enough text preprocessing can create many new features
```

Comment:
Does 20 newsgroup work in the setting on the left? That would be preferable for running this example in the github actions.

Reply:
Maybe we should also use a smaller dataset? You can use the following script to scan on OpenML for datasets containing string data:
```
import openml

datasets = openml.datasets.list_datasets()
for did in datasets:
    try:
        dataset = openml.datasets.get_dataset(did, download_data=False, download_qualities=False)
        for feat in dataset.features:
            if dataset.features[feat].data_type == 'string':
                print(did)
                break
    except Exception as e:
        print(e)
        continue
```

Reply:
the example yields ~80% acc. on the test set. Selecting random would be  5% for 20 labels. Therefore i would say that the example works. But it also runs 300 sec. which are 5 min. So if that is to long i can search another dataset.

Reply:
Sorry, I meant, would the example work when you restrict it to use only a single configuration?

Reply:
is there a parameter for setting autosklearn to it or is that max_time == timer per model ?

Reply:
I would read through the entire API and manual now that you have a bit more familiarity, to know what's possible and what's not
https://automl.github.io/auto-sklearn/master/api.html

Reply:
It has been there in the previous version of the example: `smac_scenario_args={"runcount_limit": 1}`